# Thanksgiving Dinner Data

In [6]:
import pandas as pd
data = pd.read_csv("thanksgiving.csv",encoding="Latin-1")

### Taking a look at the data

In [5]:
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


### Which columns do we have?

In [4]:
print(data.columns)

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

### How many people celebrate Thanksgiving?

In [3]:
data["Do you celebrate Thanksgiving?"].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

### We don't want date about people who don't celebrate Thanksgiving, let's focus on the ones who do.

In [12]:
data = data[data["Do you celebrate Thanksgiving?"]=="Yes"]

## Analysing the main dishes people tend to eat during Thanksgiving dinner

### Which are the main dishes and how popular are they?

In [13]:
data["What is typically the main dish at your Thanksgiving dinner?"].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

### Let's see if people who eat Tufurkey usually have gravy...

In [26]:
data[data["What is typically the main dish at your Thanksgiving dinner?"]=="Tofurkey"]["Do you typically have gravy?"].value_counts()

Yes    12
No      8
Name: Do you typically have gravy?, dtype: int64

## Exploring the dessert dishes

### Specifically, we'll look at how many people eat Apple, Pecan, or Pumpkin pie during Thanksgiving dinner. 

In [27]:
apple_isnull = pd.isnull(data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple"])

pumpkin_isnull = pd.isnull(data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin"])

pecan_isnull = pd.isnull(data["Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"])

ate_pies = apple_isnull & pumpkin_isnull & pecan_isnull

ate_pies.value_counts()

False    876
True     104
dtype: int64

###It looks that 104 people typically serve one (or more!) of these 3 pies.

## Analysing ages

In [20]:
data["Age"].value_counts()


45 - 59    269
60+        258
30 - 44    235
18 - 29    185
Name: Age, dtype: int64

### Let's analyze the Age column in more depth, but first need to convert it to numeric values. This will make it simple to figure out things like the average age of survey respondents.

### Let's create a function to convert a single string to an appropriate integer value. This will allow us to convert the values in the Age column to integers.

In [22]:
def extract_age(age_str):
    if pd.isnull(age_str):
        return None
    age_str = age_str.split(" ")[0]
    age_str = age_str.replace("+", "")
    return int(age_str)


In [23]:
data["int_age"] = data["Age"].apply(extract_age)
data["int_age"].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

### We only have a approximation of age, and it's lower than reality because we took the first value in each string.

## Analysing the incomes

In [24]:
data["How much total combined money did all members of your HOUSEHOLD earn last year?"].value_counts()

$25,000 to $49,999      166
$75,000 to $99,999      127
$50,000 to $74,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

### We have to convert these values to numeric again, like we did it with the "Age" column.

In [25]:
def extract_income (income_str):
    if pd.isnull(income_str):
        return None
    income_str = income_str.split(" ")[0]
    if income_str == "Prefer":
        return None
    income_str = income_str.replace("$", "")
    income_str = income_str.replace(",", "")
    income_str = int(income_str)
    return income_str

data["int_income"] = data["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(extract_income)
data["int_income"].describe()

count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64

### As in the last convertion, it's only an approximation of income (the lower bound). The average income seems to be high, although there is also a large standard deviation

## Analysing travelings on Thanksgiving

In [22]:
data[data["int_income"]<150000]["How far will you travel for Thanksgiving?"].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
Name: How far will you travel for Thanksgiving?, dtype: int64

In [23]:
data[data["int_income"]>150000]["How far will you travel for Thanksgiving?"].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         49
Thanksgiving is local--it will take place in the town I live in                     25
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    16
Thanksgiving is out of town and far away--I have to drive several hours or fly      12
Name: How far will you travel for Thanksgiving?, dtype: int64

### Appearently, more people with higher incomes will stay home during Thanksgiving than the ones with lower incomes. Why? Maybe a great part of the second population is student, so the will go home (out of town).

## Analysing questions related to friendship

In [32]:
data.pivot_table(
    index="Have you ever tried to meet up with hometown friends on Thanksgiving night?", 
    columns='Have you ever attended a "Friendsgiving?"',
    values="int_age"
)

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [33]:
data.pivot_table(
    index="Have you ever tried to meet up with hometown friends on Thanksgiving night?", 
    columns='Have you ever attended a "Friendsgiving?"',
    values="int_income"
)

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

### We can conclude that people who are younger or have lower incomes are more likely to attend a Friendsgiving and meet up with friends on Thanksgiving night.